workflows/validate.py

```diff
@@ -0,0 +1,78 @@
+"""
+Runs the json validator on a json file and raises an exception if there is an error
+Intended to be used as a hook for repositories which are developing recipes
+
+Example of how this could be used in a .pre-commit-config.yaml file:
+```
+# Validate json recipes
+- repo: https://github.com/DiamondLightSource/python-workflows
+  hooks:
+  - name: Recipe validation
+    id: recipe-validation
+    language: python
+    files: ^recipes/
+    entry: workflows-recipe-validate
```

Comment:
We should declare the pre-commit hook in workflows so people can use it easily in their repositories.
Also I think the file pattern should be `*.json` rather than `recipes/`

Reply:
What do you mean by declare? Is there a method to make it easier for people to use the hook? That makes sense as well, but could create a headache if you have other jsons floating around for configuration/other purposes?

Reply:
You can declare this repository to contain pre-commit hooks via a `.pre-commit-hooks.yaml` file, which can then be referred to in other repositories using the usual `.pre-commit-config.yaml`. See ["Creating new hooks"](https://pre-commit.com/#new-hooks), and eg. https://github.com/python/black/blob/master/.pre-commit-hooks.yaml for an example.

Reply:
Cool, will add that. Further thoughts on *.json vs /recipes?

Reply:
Added this with types by default and files commented out as a suggestion (good practice might be to use both ?)
`yaml
- id: workflows-recipe-validation
  name: Workflows Recipe Validation
  description: Ensure recipes designed to work with python-workflows are compliant
  entry: workflows.validate_recipe
  language: python
  #files: ^recipes/
  types: [json]
`